typeclasses/chargen.py

```diff
@@ -0,0 +1,587 @@
+"""
+Chargen EvMenu module.
+
+The menu node functions defined in this module make up
+the Ainneve character creation process, which is based
+on a subset of Open Adventure rules.
+"""
+from evennia import spawn
+from evennia.utils import fill, dedent
+from evennia.utils.evtable import EvTable
+
+from world import archetypes, races, skills
+from world.rulebook import d_roll
+from world.economy import format_coin as as_price
+from world.economy import transfer_funds, InsufficientFunds
+
+from math import ceil
+import re
+
+
+# Organize starter equipment prototypes by category
+_EQUIPMENT_CATEGORIES = {
+    'Melee Weapons':
+        (0, 'HAND_AXE', 'BATTLE_AXE', 'MAUL_HAMMER', 'LANCE_POLEARM',
+         'PIKE_POLEARM', 'MAPLE_STAFF', 'MACE_ROD', 'MORNINGSTAR_ROD',
+         'SCYTHE', 'SHORT_SWORD', 'RAPIER', 'WHIP'),
+    'Ranged Weapons':
+        (1, 'LONG_BOW', 'HAND_CROSSBOW', 'LIGHT_CROSSBOW'),
+    'Ammunition':
+        (5, 'ARROW_BUNDLE', 'QUARREL_BUNDLE'),
+    'Thrown Weapons':
+        (2, 'THROWING_AXE', 'THROWING_DAGGER', 'JAVELIN', 'TRIDENT'),
+    'Armor':
+        (3, 'CLOTH_GARMENT', 'LEATHER_GARMENT', 'BRIGANDINE', 'CHAIN_MAIL',
+         'PLATED_MAIL', 'IRON_SCALE', 'IRON_BANDED'),
+    'Shields':
+        (4, 'BUCKLER_SHIELD', 'HERALDIC_SHIELD', 'TOWER_SHIELD'),
+}
+
+
+_CATEGORY_LIST = sorted(_EQUIPMENT_CATEGORIES.iterkeys(),
+                        key=lambda c: _EQUIPMENT_CATEGORIES[c][0])
+
+
+_RE_ARTCL = re.compile(r'^an?\s', re.IGNORECASE)
+
+
+def menunode_welcome_archetypes(caller):
+    """Starting page and Archetype listing."""
+    text = dedent("""\
+        |wWelcome to |mAinneve|w, the example game for |yEvennia|w.|n
+
+        To begin, select an |cArchetype|n. There are three base
+        archetypes to choose from, plus three dual archetypes which
+        combine the strengths and weaknesses of two archetypes into one.
+
+        Select an Archetype by number below to view its details, or "help" for more info.
+    """)
+    help = fill("In |mAinneve|n, character |cArchetypes|n represent the "
+                "characters' class or primary role. The various archetypes "
+                "have different strengths and weaknesses, which are reflected "
+                "in your character's starting traits.")
+    options = []
+    for arch in archetypes.VALID_ARCHETYPES:
+        a = archetypes.load_archetype(arch)
+        options.append({"desc": "|c{}|n".format(a.name),
+                        "goto": "menunode_select_archetype"})
+    return (text, help), options
+
+
+def menunode_select_archetype(caller, raw_input):
```

Comment:
I think we should change the `raw_input` parameter to `raw_string`, which avoids potentially confusing newbies with Python's raw_input function and follows EvMenu's own naming convention.